### ISSUE TITLE
Conv/pool doc on ceilmode wrong

### ISSUE THREAD

**Issue by albertz** [pytorch/pytorch#148123]
### 📚 The doc issue

The doc on maxpool/conv states this output shape:
$$L_{out} = \left\lfloor \frac{L_{in} + 2 \times \text{padding} - \text{dilation}  \times (\text{kernelsize} - 1) - 1}{\text{stride}} + 1\right\rfloor$$
Then it says for the `ceil_mode` arg: "If True, will use ceil instead of floor to compute the output shape"
This doc on `ceil_mode` seems to be wrong, or misleading, I'm not sure. I would guess it would mean:
$$L_{out}^{\text{ceilmode}} = \left\lceil \frac{L_{in} + 2 \times \text{padding} - \text{dilation}  \times (\text{kernelsize} - 1) - 1}{\text{stride}} + 1\right\rceil$$

For `padding=0`, `dilation=1`, `kernel_size=1`, `stride=2`:
* For `L_in=10`, `ceil_mode=False`: The formula gives `L_out=5`. That's the case.
* For `L_in=11`, `ceil_mode=False`: The formula gives `L_out=6`. That's the case.
* For `L_in=10`, `ceil_mode=True`: The formula gives `L_out=6`. But that's not the case: `torch.nn.functional.max_pool1d(torch.zeros(1,1,10),1,2,ceil_mode=True).shape` gives `torch.Size([1, 1, 5])`.

So, `ceil_mode` seems to mean sth different. But what exactly?

Btw, related to that, the option name is anyway confusing to me. Because the original formula is already like ceil-mode to me, because you have this identity:
$$L_{out} = \left\lfloor \frac{L_{in} + 2 \times \text{padding} - \text{dilation}  \times (\text{kernelsize} - 1) - 1}{\text{stride}} + 1\right\rfloor = \left\lfloor \frac{L_{in} + 2 \times \text{padding} - \text{dilation}  \times (\text{kernelsize} - 1) + (\text{stride} - 1)}{\text{stride}}\right\rfloor  = \left\lceil \frac{L_{in} + 2 \times \text{padding} - \text{dilation}  \times (\text{kernelsize} - 1)}{\text{stride}}\right\rceil$$
I find this actually a much more natural way to write this and to think about it. But when written like that, what does `ceil_mode` mean now?

More specifically, how exactly is the formula of `L_out` with `ceil_mode=True`?

### Suggest a potential alternative/fix

Clarify/reword the doc of `ceil_mode`, and add the exact formula of the output shape for `ceil_mode=True`.

**Comment by albertz** [comment]
Can this be reopened? There is a bug in the doc now. The PR links this code:

https://github.com/pytorch/pytorch/blob/00199acdb85a4355612bff28e1018b035e0e46b9/aten/src/ATen/native/Pool.h#L61-L75

So in any case, it stays floor division, when you write it like that.

But anyway, also see my original comment. The terminology is anyway confusing. Because even with `ceil_mode=False`, this is actually better (more natural) described in terms of ceil-division.

Every floor division can be rewritten as ceil division, and vice versa.

(cc @drisspg @zeshengzong @AmdSampsa)
